9-Tools/configfile_generator/ConstructAccess.py

```diff
@@ -0,0 +1,225 @@
+	create_config(namespace_name,namespace_username)
+
+	# Cleaning up 
+	exec_get_output(["rm","-rf","sa.yaml", "role.yaml", "rolebinding.yaml"])
+
+
+# Given the config files this will merge everything and output it to "KUBECONFIG"
+def merge_configs(files):
+	cmd = "export KUBECONFIG=mergedconfig"
+	for name in files:
+		cmd = cmd + ":" + name
+	os.environ["KUBECONFIG"] = cmd
+	cmd = "kubectl config view --flatten  > mergedconfig"
+	exec_get_output(cmd,True)
+	os.environ["KUBECONFIG"] = "$HOME/.kube/config"
+
+
+# Given a resource quota yaml file, apply it on a namespace using kubectl.
+def limit_resources(resourcelimitfilepath,namespace_name):
+	# we need to change the namespace in the yaml file for each namespace 
+	# so that the same file can be applied
+	print(exec_get_output("kubectl apply -f " + resourcelimitfilepath + " -n " + namespace_name,True))
+
+	exec_get_output("cp " + resourcelimitfilepath + " ./" + namespace_name,True)
+
+
+# Syntax "python ConstructAccess.py merge file1 file2 file3"
+def case1():
+	merge_configs(list(sys.argv[2:]))
+
+
+# Syntax "python ConstructAccess.py create --akind access_kind namespace1 namespace2 ... namespaceN"
+def case2():
+	type(optsdict)
+	for elem in args:
+		namespace_username = elem + "-user"
+		generate_new_config(elem,namespace_username,sys.argv[1],optsdict['--akind'])
+
+
+# Syntax "python ConstructAccess.py createEx --nsname namespace_name --akind access_kind username1  ... usernameN"
+def case3():
+	for elem in args:
+		generate_new_config(optsdict['--nsname'],elem,sys.argv[1],optsdict['--akind'])
+
+
+# Syntax "python ConstructAccess.py createCustomRole --rpath role_file_path namespace1 namespace2 ... namepspaceN"
+def case4():
+	for elem in args:
+		namespace_username = elem + "-user"
+		generate_new_config(elem,namespace_username,sys.argv[1],role_file_path=optsdict['--rpath'])
+
+
+# Syntax "python ConstructAccess.py createExCustomeRole --nsname namespace_name --rpath role_file_path username1 username2 .... usernameN"
+def case5():
+	for elem in args:
+		generate_new_config(optsdict['--nsname'],elem,sys.argv[1],role_file_path=optsdict['--rpath'])
+
+
+# Syntax "python ConstructAccess.py recreate --nsname namespace_name username1 username2 username3 ... usernameN"
+def case6():
+	for elem in args:
+		create_config(sys.argv[2],elem)
+
+
+# Syntax "python ConstructAccess.py limitRes --rlpath ResourceQuotafilepath namespace1 ... namespaceN"
+def case7():
+	print(args)
+	for elem in args:
+		limit_resources(optsdict['--rlpath'],elem)
+
+
+# all flags and their shorthands, 
+# --nsname -a
+# --uname -b
+# --akind -c
+# --rpath -d
+# -rlpath -e
+if __name__ == '__main__':
+	cases = {"merge":case1,"create":case2,"createEx":case3,"createCustomRole":case4,
+	"createExCustomRole":case5,"recreate":case6,"limitRes":case7}
+	try:
+		opts, args = getopt.getopt(sys.argv[2:], 'a:b:c:d:e:',['nsname=', 'uname=','akind=','rpath=','rlpath='])
```

Comment:
These short flags are not well readable. I would suggest to remove them, or use some letters which are relevant to their functionalities. For example, --nsname could be -n, --uname could be -u.

Reply:
I changed them to relevant letters :).